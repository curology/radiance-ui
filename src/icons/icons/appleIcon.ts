import { AppleIcon } from './svgs';
import { IconProps, useIcon } from '../../shared-components/icon';

export default (props: IconProps) => useIcon(AppleIcon, AppleIcon, props);
