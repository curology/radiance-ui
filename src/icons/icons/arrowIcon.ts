import { ArrowIcon } from './svgs';
import { IconProps, useIcon } from '../../shared-components/icon';

export default (props: IconProps) => useIcon(ArrowIcon, ArrowIcon, props);
