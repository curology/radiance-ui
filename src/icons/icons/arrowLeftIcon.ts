import { ArrowLeftIcon, ArrowLeftIconSecondary } from './svgs';
import { IconProps, useIcon } from '../../shared-components/icon';

export default (props: IconProps) =>
  useIcon(ArrowLeftIcon, ArrowLeftIconSecondary, props);
