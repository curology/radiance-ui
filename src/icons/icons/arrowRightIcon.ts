import { ArrowRightIcon, ArrowRightIconSecondary } from './svgs';
import { IconProps, useIcon } from '../../shared-components/icon';

export default (props: IconProps) =>
  useIcon(ArrowRightIcon, ArrowRightIconSecondary, props);
