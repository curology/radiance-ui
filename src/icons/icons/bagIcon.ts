import { BagIcon, BagIconSecondary } from './svgs';
import { IconProps, useIcon } from '../../shared-components/icon';

export default (props: IconProps) => useIcon(BagIcon, BagIconSecondary, props);
