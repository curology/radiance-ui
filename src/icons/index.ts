/*
 * Note: this file has to be manually updated if we add more types of icons.
 * We do not generate it via create-index
 */

export * from './emojis';
export * from './glyphs';
export * from './icons';
export * from './logos';
export * from './navIcons';
