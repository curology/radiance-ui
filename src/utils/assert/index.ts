import { isDefined } from '../isDefined';

function assert(value: boolean, message?: string): asserts value;
function assert<T>(
  value: T | null | undefined,
  message?: string,
): asserts value is T;
/**
 * @see https://gist.github.com/jamiebuilds/2a40f565cba317a80bed1eec127340f6
 */
function assert(value: unknown, message?: string) {
  if (value === false || !isDefined(value)) {
    throw new Error(message ?? 'Assertion failed');
  }
}

export { assert };
