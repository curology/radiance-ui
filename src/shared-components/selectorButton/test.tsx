import React from 'react';
import { render, userEvent } from 'src/tests/testingLibraryHelpers';
import { assert } from 'src/utils/assert';

import { AcneOneGlyph } from '../../icons';

import { SelectorButton } from './index';

describe('<SelectorButton />', () => {
  describe('UI snapshots', () => {
    it('when children is undefined', () => {
      const { container } = render(
        <SelectorButton checked={false} onClick={() => undefined} />,
      );

      expect(container.firstElementChild).toMatchSnapshot();
    });

    it('when children is a node', () => {
      const { container } = render(
        <SelectorButton checked={false} onClick={() => undefined}>
          SelectorButton Text
        </SelectorButton>,
      );

      expect(container.firstElementChild).toMatchSnapshot();
    });

    it('when checked type is primary', () => {
      const { container } = render(
        <SelectorButton checked onClick={() => undefined} type="primary">
          SelectorButton Text
        </SelectorButton>,
      );

      expect(container.firstElementChild).toMatchSnapshot();
    });

    it('when checked type is secondary', () => {
      const { container } = render(
        <SelectorButton checked onClick={() => undefined} type="secondary">
          SelectorButton Text
        </SelectorButton>,
      );

      expect(container.firstElementChild).toMatchSnapshot();
    });

    it('when is checkbox', () => {
      const { container } = render(
        <SelectorButton
          checked={false}
          onClick={() => undefined}
          selector="checkbox"
        >
          SelectorButton Text
        </SelectorButton>,
      );

      expect(container.firstElementChild).toMatchSnapshot();
    });

    describe('when Icon added', () => {
      it("hides icon for checkbox with size 'small'", () => {
        const { container } = render(
          <SelectorButton
            checked={false}
            onClick={() => undefined}
            selector="checkbox"
            icon={<AcneOneGlyph />}
            size="small"
          >
            SelectorButton Text
          </SelectorButton>,
        );

        expect(container.querySelectorAll('svg')).toHaveLength(0);
      });

      it("hides icon for radio button size 'small'", () => {
        const { container } = render(
          <SelectorButton
            checked={false}
            onClick={() => undefined}
            selector="radio"
            icon={<AcneOneGlyph />}
            size="small"
          >
            SelectorButton Text
          </SelectorButton>,
        );

        expect(container.querySelectorAll('svg')).toHaveLength(0);
      });

      it("displays icon for checkbox with size 'large'", () => {
        const { container } = render(
          <SelectorButton
            checked={false}
            onClick={() => undefined}
            selector="checkbox"
            icon={<AcneOneGlyph />}
            size="large"
          >
            SelectorButton Text
          </SelectorButton>,
        );

        expect(container.querySelectorAll('svg')).toHaveLength(1);
      });

      it("displays icon for radio button with size 'large'", () => {
        const { container } = render(
          <SelectorButton
            checked={false}
            onClick={() => undefined}
            selector="radio"
            icon={<AcneOneGlyph />}
            size="large"
          >
            SelectorButton Text
          </SelectorButton>,
        );

        expect(container.querySelectorAll('svg')).toHaveLength(1);
      });

      it('displays check mark for checked checkbox', () => {
        const { container } = render(
          <SelectorButton
            checked
            onClick={() => undefined}
            selector="checkbox"
            icon={<AcneOneGlyph />}
            size="large"
          >
            SelectorButton Text
          </SelectorButton>,
        );

        expect(container.firstElementChild).toMatchSnapshot();
      });

      it('displays check mark for checked radio button', () => {
        const { container } = render(
          <SelectorButton
            checked
            onClick={() => undefined}
            selector="radio"
            icon={<AcneOneGlyph />}
            size="large"
          >
            SelectorButton Text
          </SelectorButton>,
        );

        expect(container.firstElementChild).toMatchSnapshot();
      });
    });
  });

  describe('onClick callback', () => {
    it('is invoked on click', () => {
      const spy = jest.fn();
      const { container } = render(
        <SelectorButton checked={false} onClick={spy} />,
      );

      assert(container.firstElementChild);
      userEvent.click(container.firstElementChild);

      expect(spy).toHaveBeenCalled();
    });

    it('does nothing when no onClick is set', () => {
      const { container } = render(<SelectorButton checked={false} />);

      // Just check that no exception is thrown
      expect(() => {
        assert(container.firstElementChild);
        userEvent.click(container.firstElementChild);
      }).not.toThrow();
      expect(() => {
        assert(container.firstElementChild);
        userEvent.type(container.firstElementChild, '{enter}');
      }).not.toThrow();
    });

    it('is invoked when enter is pressed', () => {
      const spy = jest.fn();
      const { container } = render(
        <SelectorButton checked={false} onClick={spy} />,
      );

      assert(container.firstElementChild);
      userEvent.type(container.firstElementChild, '{enter}');

      expect(spy).toHaveBeenCalled();
    });
  });
});
