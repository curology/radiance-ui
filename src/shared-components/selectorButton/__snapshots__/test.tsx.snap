// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectorButton /> UI snapshots when Icon added displays check mark for checked checkbox 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  margin-bottom: 20px;
}

.emotion-0:focus {
  outline: none;
}

.emotion-0:focus .emotion-3 {
  box-shadow: 0px 0px 0px 2px #FFFFFF,0px 0px 0px 4px #332E54;
  border-radius: 4px;
}

.emotion-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
}

.emotion-4 {
  fill: currentColor;
  left: 50%;
  position: absolute;
  top: 50%;
  -webkit-transform: translate(-50%, -50%);
  -moz-transform: translate(-50%, -50%);
  -ms-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
  cursor: pointer;
}

.emotion-6 {
  display: block;
  -webkit-transform: rotate(0deg);
  -moz-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: color 350ms,-webkit-transform 350ms;
  transition: color 350ms,transform 350ms;
  color: #FFFFFF;
}

.emotion-7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  border: 1px solid;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 3rem;
  height: 3rem;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-transition: background-color 350ms;
  transition: background-color 350ms;
  border-radius: 4px;
  background-color: #332E54;
  border-color: #332E54;
}

.emotion-7:active,
.emotion-7:focus {
  outline: none;
}

.emotion-9 {
  margin-left: 1rem;
  margin-top: 0.25rem;
  min-width: 125px;
  text-align: left;
}

<div
  aria-checked="true"
  class="emotion-0 emotion-1"
  role="checkbox"
  tabindex="0"
>
  <div
    class="emotion-2 emotion-3"
  >
    <div
      class="emotion-4 emotion-5"
    >
      <svg
        class="emotion-6"
        fill="none"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14.96 2l-9.7 9.42L.88 7.1 0 7.99l5.26 5.18L15.83 2.9l-.87-.9z"
          fill="currentColor"
        />
      </svg>
    </div>
    <div
      class="emotion-7 emotion-8"
      type="primary"
    />
  </div>
  <p
    class="emotion-9 emotion-10"
  >
    SelectorButton Text
  </p>
</div>
`;

exports[`<SelectorButton /> UI snapshots when Icon added displays check mark for checked radio button 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  margin-bottom: 20px;
}

.emotion-0:focus {
  outline: none;
}

.emotion-0:focus .emotion-3 {
  box-shadow: 0px 0px 0px 2px #FFFFFF,0px 0px 0px 4px #332E54;
  border-radius: 50%;
}

.emotion-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
}

.emotion-4 {
  fill: currentColor;
  left: 50%;
  position: absolute;
  top: 50%;
  -webkit-transform: translate(-50%, -50%);
  -moz-transform: translate(-50%, -50%);
  -ms-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
  cursor: pointer;
}

.emotion-6 {
  display: block;
  -webkit-transform: rotate(0deg);
  -moz-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: color 350ms,-webkit-transform 350ms;
  transition: color 350ms,transform 350ms;
  color: #FFFFFF;
}

.emotion-7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  border: 1px solid;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 3rem;
  height: 3rem;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-transition: background-color 350ms;
  transition: background-color 350ms;
  border-radius: 50%;
  background-color: #332E54;
  border-color: #332E54;
}

.emotion-7:active,
.emotion-7:focus {
  outline: none;
}

.emotion-9 {
  margin-left: 1rem;
  margin-top: 0.25rem;
  min-width: 125px;
  text-align: left;
}

<div
  aria-checked="true"
  class="emotion-0 emotion-1"
  role="radio"
  tabindex="0"
>
  <div
    class="emotion-2 emotion-3"
  >
    <div
      class="emotion-4 emotion-5"
    >
      <svg
        class="emotion-6"
        fill="none"
        height="8"
        viewBox="0 0 16 16"
        width="8"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="8"
          cy="8"
          fill="currentColor"
          r="7.4"
          stroke="currentColor"
          stroke-width="1.2"
        />
      </svg>
    </div>
    <div
      class="emotion-7 emotion-8"
      type="primary"
    />
  </div>
  <p
    class="emotion-9 emotion-10"
  >
    SelectorButton Text
  </p>
</div>
`;

exports[`<SelectorButton /> UI snapshots when checked type is primary 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  margin-bottom: 20px;
}

.emotion-0:focus {
  outline: none;
}

.emotion-0:focus .emotion-3 {
  box-shadow: 0px 0px 0px 2px #FFFFFF,0px 0px 0px 4px #332E54;
  border-radius: 50%;
}

.emotion-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
}

.emotion-4 {
  fill: currentColor;
  left: 50%;
  position: absolute;
  top: 50%;
  -webkit-transform: translate(-50%, -50%);
  -moz-transform: translate(-50%, -50%);
  -ms-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
  cursor: pointer;
}

.emotion-6 {
  display: block;
  -webkit-transform: rotate(0deg);
  -moz-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: color 350ms,-webkit-transform 350ms;
  transition: color 350ms,transform 350ms;
  color: #FFFFFF;
}

.emotion-7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  border: 1px solid;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 2rem;
  height: 2rem;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-transition: background-color 350ms;
  transition: background-color 350ms;
  border-radius: 50%;
  background-color: #332E54;
  border-color: #332E54;
}

.emotion-7:active,
.emotion-7:focus {
  outline: none;
}

.emotion-9 {
  margin-left: 1rem;
  margin-top: 0.25rem;
  min-width: 125px;
  text-align: left;
}

<div
  aria-checked="true"
  class="emotion-0 emotion-1"
  role="radio"
  tabindex="0"
>
  <div
    class="emotion-2 emotion-3"
  >
    <div
      class="emotion-4 emotion-5"
    >
      <svg
        class="emotion-6"
        fill="none"
        height="8"
        viewBox="0 0 16 16"
        width="8"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="8"
          cy="8"
          fill="currentColor"
          r="7.4"
          stroke="currentColor"
          stroke-width="1.2"
        />
      </svg>
    </div>
    <div
      class="emotion-7 emotion-8"
      type="primary"
    />
  </div>
  <p
    class="emotion-9 emotion-10"
  >
    SelectorButton Text
  </p>
</div>
`;

exports[`<SelectorButton /> UI snapshots when checked type is secondary 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  margin-bottom: 20px;
}

.emotion-0:focus {
  outline: none;
}

.emotion-0:focus .emotion-3 {
  box-shadow: 0px 0px 0px 2px #FFFFFF,0px 0px 0px 4px #332E54;
  border-radius: 50%;
}

.emotion-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
}

.emotion-4 {
  fill: currentColor;
  left: 50%;
  position: absolute;
  top: 50%;
  -webkit-transform: translate(-50%, -50%);
  -moz-transform: translate(-50%, -50%);
  -ms-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
  cursor: pointer;
}

.emotion-6 {
  display: block;
  -webkit-transform: rotate(0deg);
  -moz-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: color 350ms,-webkit-transform 350ms;
  transition: color 350ms,transform 350ms;
  color: #FFFFFF;
}

.emotion-7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  border: 1px solid;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 2rem;
  height: 2rem;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-transition: background-color 350ms;
  transition: background-color 350ms;
  border-radius: 50%;
  background-color: #A6A1E2;
  border-color: #A6A1E2;
}

.emotion-7:active,
.emotion-7:focus {
  outline: none;
}

.emotion-9 {
  margin-left: 1rem;
  margin-top: 0.25rem;
  min-width: 125px;
  text-align: left;
}

<div
  aria-checked="true"
  class="emotion-0 emotion-1"
  role="radio"
  tabindex="0"
>
  <div
    class="emotion-2 emotion-3"
  >
    <div
      class="emotion-4 emotion-5"
    >
      <svg
        class="emotion-6"
        fill="none"
        height="8"
        viewBox="0 0 16 16"
        width="8"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="8"
          cy="8"
          fill="currentColor"
          r="7.4"
          stroke="currentColor"
          stroke-width="1.2"
        />
      </svg>
    </div>
    <div
      class="emotion-7 emotion-8"
      type="secondary"
    />
  </div>
  <p
    class="emotion-9 emotion-10"
  >
    SelectorButton Text
  </p>
</div>
`;

exports[`<SelectorButton /> UI snapshots when children is a node 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  margin-bottom: 20px;
}

.emotion-0:focus {
  outline: none;
}

.emotion-0:focus .emotion-3 {
  box-shadow: 0px 0px 0px 2px #FFFFFF,0px 0px 0px 4px #332E54;
  border-radius: 50%;
}

.emotion-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
}

.emotion-4 {
  fill: currentColor;
  left: 50%;
  position: absolute;
  top: 50%;
  -webkit-transform: translate(-50%, -50%);
  -moz-transform: translate(-50%, -50%);
  -ms-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
  cursor: pointer;
}

.emotion-6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  border: 1px solid;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 2rem;
  height: 2rem;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-transition: background-color 350ms;
  transition: background-color 350ms;
  border-radius: 50%;
  background-color: transparent;
  border-color: #332E54;
}

.emotion-6:active,
.emotion-6:focus {
  outline: none;
}

.emotion-8 {
  margin-left: 1rem;
  margin-top: 0.25rem;
  min-width: 125px;
  text-align: left;
}

<div
  aria-checked="false"
  class="emotion-0 emotion-1"
  role="radio"
  tabindex="0"
>
  <div
    class="emotion-2 emotion-3"
  >
    <div
      class="emotion-4 emotion-5"
    />
    <div
      class="emotion-6 emotion-7"
      type="primary"
    />
  </div>
  <p
    class="emotion-8 emotion-9"
  >
    SelectorButton Text
  </p>
</div>
`;

exports[`<SelectorButton /> UI snapshots when children is undefined 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  margin-bottom: 20px;
}

.emotion-0:focus {
  outline: none;
}

.emotion-0:focus .emotion-3 {
  box-shadow: 0px 0px 0px 2px #FFFFFF,0px 0px 0px 4px #332E54;
  border-radius: 50%;
}

.emotion-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
}

.emotion-4 {
  fill: currentColor;
  left: 50%;
  position: absolute;
  top: 50%;
  -webkit-transform: translate(-50%, -50%);
  -moz-transform: translate(-50%, -50%);
  -ms-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
  cursor: pointer;
}

.emotion-6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  border: 1px solid;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 2rem;
  height: 2rem;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-transition: background-color 350ms;
  transition: background-color 350ms;
  border-radius: 50%;
  background-color: transparent;
  border-color: #332E54;
}

.emotion-6:active,
.emotion-6:focus {
  outline: none;
}

<div
  aria-checked="false"
  class="emotion-0 emotion-1"
  role="radio"
  tabindex="0"
>
  <div
    class="emotion-2 emotion-3"
  >
    <div
      class="emotion-4 emotion-5"
    />
    <div
      class="emotion-6 emotion-7"
      type="primary"
    />
  </div>
</div>
`;

exports[`<SelectorButton /> UI snapshots when is checkbox 1`] = `
.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  margin-bottom: 20px;
}

.emotion-0:focus {
  outline: none;
}

.emotion-0:focus .emotion-3 {
  box-shadow: 0px 0px 0px 2px #FFFFFF,0px 0px 0px 4px #332E54;
  border-radius: 4px;
}

.emotion-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
}

.emotion-4 {
  fill: currentColor;
  left: 50%;
  position: absolute;
  top: 50%;
  -webkit-transform: translate(-50%, -50%);
  -moz-transform: translate(-50%, -50%);
  -ms-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
  cursor: pointer;
}

.emotion-6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  border: 1px solid;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 2rem;
  height: 2rem;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-transition: background-color 350ms;
  transition: background-color 350ms;
  border-radius: 4px;
  background-color: transparent;
  border-color: #332E54;
}

.emotion-6:active,
.emotion-6:focus {
  outline: none;
}

.emotion-8 {
  margin-left: 1rem;
  margin-top: 0.25rem;
  min-width: 125px;
  text-align: left;
}

<div
  aria-checked="false"
  class="emotion-0 emotion-1"
  role="checkbox"
  tabindex="0"
>
  <div
    class="emotion-2 emotion-3"
  >
    <div
      class="emotion-4 emotion-5"
    />
    <div
      class="emotion-6 emotion-7"
      type="primary"
    />
  </div>
  <p
    class="emotion-8 emotion-9"
  >
    SelectorButton Text
  </p>
</div>
`;
